api/prometheus/v1/api_test.go

```diff
@@ -15,17 +15,20 @@ package v1
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
 	"fmt"
+	"math"
 	"net/http"
 	"net/url"
 	"reflect"
 	"strings"
 	"testing"
 	"time"
 
+	json "github.com/json-iterator/go"
+
 	"github.com/prometheus/common/model"
+	"github.com/prometheus/tsdb/testutil"
```

Comment:
Not part of this PR (as it happens in api/client_test.go already), but now that I'm seeing it here:

I find testutil packages problematic. The usual Go mantra is to not introduce custom test utilities and frameworks. Saving a few repetitive keystrokes comes at the cost of more dependencies and cognitive overhead for the reader (who has to adjust to the many different testutil packages out there). Using it in `prometheus/tsdb` might be justified as it is quite self-contained code. `prometheus/client_golang`, however, is deliberately meant to be used by many users with different stylistic ideas. The cognitive overhead for reading the code and the added dependency has more of an impact here. Note that the `prometheus/client_golang` has a dependency on `prometheus/tsdb` only because of the usage of `testutil`. Also note that this `testutil` package is not in `prometheus/common`. It was probably never meant to be used outside of `prometheus/tsdb`.

I'd like to see that dependency gone, but I'll open a separate issue for this.

Reply:
Thats unfortunate because thats what I had originally (no testutil). If this is something we want going forward we should be clear about it up-front :)

Reply:
Yes, we definitely should.

If we want to deviate from common Go idioms and mantras, we should have an informed consensus about it among the Prometheus developers.

Reply:
I've re-re-re written the tests, this time (again) to not use testutil :)

Reply:
>  The cognitive overhead for reading the code and the added dependency has more of an impact here. Note that the prometheus/client_golang has a dependency on prometheus/tsdb only because of the usage of testutil. 

In general I agree, but in this case I think it is exactly the opposite. It simplifies the code without any cognitive overhead.

In general it would be ideal to have in common and use across projects.